contracts/JoeMakerV4.sol

```diff
@@ -260,24 +329,50 @@ contract JoeMakerV4 is Ownable {
         uint256 amountInput = IERC20(fromToken).balanceOf(address(pair)).sub(reserveInput); // calculate amount that was transferred, this accounts for transfer taxes
 
         amountOut = getAmountOut(amountInput, reserveInput, reserveOutput);
+
+        {
+            uint256 rest = uint256(10_000).sub(slippage);
+            /// @dev We calculate the amount received if we did a swapIn and swapOut, without updating the reserves,
+            /// this allows to catch if a pair has low liquidity. We do `rest^2`, i.e. calculating twice the slippage
+            /// cause we actually do two swaps, so slippage is actually applied twice.
+            require(
+                getAmountOut(amountOut, reserveOutput, reserveInput) >=
+                    amountInput.mul(rest).mul(rest).div(100_000_000),
+                "JoeMakerV4: Slippage caught"
+            );
+        }
+
         (uint256 amount0Out, uint256 amount1Out) = fromToken == pair.token0()
             ? (uint256(0), amountOut)
             : (amountOut, uint256(0));
         pair.swap(amount0Out, amount1Out, to, new bytes(0));
     }
 
-    // F1 - F10: OK
-    // C1 - C24: OK
-    function _toToken(address token, uint256 amountIn) internal returns (uint256 amountOut) {
+    /// @notice Swaps `amountIn` `token` to `tokenTo`, `amountOut` is required to be greater
+    /// than allowed `slippage`.
```

Comment:
I'd avoid using too many variables in the comment; it's important to explain the function in simple english.

```
/// @notice Swaps an amount of token to another token, `tokenTo`
/// @dev `amountOut` is required to be greater after slippage 
```